src/NSwag.CodeGeneration/DefaultParameterNameGenerator.cs

```diff
@@ -21,19 +21,22 @@ public class DefaultParameterNameGenerator : IParameterNameGenerator
         /// <returns>The parameter name.</returns>
         public string Generate(OpenApiParameter parameter, IEnumerable<OpenApiParameter> allParameters)
         {
-            if (string.IsNullOrEmpty(parameter.Name))
+            var name = !string.IsNullOrEmpty(parameter.CustomName) ? 
```

Comment:
Didn't understand the question...

Reply:
I get it now. I don't think it is necessary if the spec parameter is named anything else than `x-name` : it is explicit that it is generated by NSwag and I'd argue that the parameter should match on both side.

What's the breaking risk of that change?

Reply:
> What's the breaking risk of that change?

Not really a break, in the current PR state you can enable/disable in the spec gen but not in the client gen - just wondering whether we should have a config for both so it's aligned.

Reply:
In the current release, what happens if I try to create an API with a different name ? Which name is taken, the param name or the HTTP one ?

The reason why we would introduce another parameter would be to make the new behavior opt-it to not break the current implementations.

Reply:
> Which name is taken, the param name or the HTTP one ?

Currently it always uses the HTTP name and the other is ignored (ie from an API perspective it doesnt matter). Currently there is no way to preserve the original parameter name.

Reply:
If I have http parameter A and C# parameter B, the name A was used in generated client code, and now it's named B.
It might be breaking in cases like these:
```cs
await client.CallMethod(A: 123);
```

I don't know if it's really a big deal though, that kind of cases should be very rare.

Do you see any other potential breaking change ?